contracts/infrastructure/dapp/CompoundFilter.sol

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2021  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.s
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+import "./IFilter.sol";
+
+contract CompoundFilter is IFilter {
```

Comment:
It seems to me that the only method that needs to be forbidden is `repayBorrowBehalf`. Everything else is safe to allow (note that ERC20 methods of cToken will be caught in `recoverSpender` and do not need to be forbidden here). In particular why not allow `borrow`, `repayBorrow`, `redeemUnderlying`?

Reply:
Agreed for the CToken methods, we should only block `reapyBorrowBehalf`.

However I believe ERC20 token methods where the spender is the cToken (i.e. the ones which will end up in the filter) must be blocked because you want to prevent e.g. a transfer of DAI to cBAT. So `dai.transfer(cBAT, 1000)` must be blocked. The only one we should accept is an approve on the underlying to allow minting, i.e. `dai.approve(cDAI,1000)` .